src/components/UI/SearchModal.vue

```diff
@@ -0,0 +1,189 @@
+        <div
+          v-for="{
+            id,
+            original_title,
+            vote_average,
+            backdrop_path,
+            title,
+            release_date,
+          } in sortedTypes.movies ? decreaseData(sortedTypes.movies) : []"
+          :key="id"
+        >
+          <search-movie-card
+            :original-title="original_title"
+            :rating="vote_average"
+            :movieImage="backdrop_path || ''"
+            :title="title"
+            :release-date="release_date ? getYear(release_date) : ''"
+            :movie-id="id"
+            @get-movie-id="getMovieDetails($event)"
+          />
+        </div>
+        <div
+          class="d-flex justify-content-end mt-2 seeMore"
+          @click="makeOptionalResponseAndRedirect('movies')"
+        >
+          see more...
+        </div>
+      </div>
+      <div v-show="sortedTypes ? sortedTypes.tvs.length !== 0 : false">
+        <h2 class="mt-2">Tvs</h2>
+        <div
+          v-for="{
+            id,
+            name,
+            original_name,
+            first_air_date,
+            vote_average,
+            poster_path,
+          } in sortedTypes.tvs ? decreaseData(sortedTypes.tvs) : []"
+          :key="id"
+        >
+          <search-tv-card
+            :name="name"
+            :original-name="original_name"
+            :release-date="first_air_date ? getYear(first_air_date) : ''"
+            :rating="vote_average"
+            :movieImage="poster_path || ''"
+            :tv-id="id"
+            @get-tv-id="getTvDetails($event)"
+          />
+        </div>
+        <div
+          class="d-flex justify-content-end mt-2 seeMore"
+          @click="makeOptionalResponseAndRedirect('tvs')"
+        >
+          see more...
+        </div>
+      </div>
+      <div v-show="sortedTypes ? sortedTypes.actors.length !== 0 : false">
+        <h2 class="mt-2">Actors</h2>
+        <div
+          v-for="{ id, name, profile_path } in sortedTypes.actors
+            ? decreaseData(sortedTypes.actors)
+            : []"
+          :key="id"
+        >
+          <search-actor-card
+            :actor-id="id"
+            :name="name"
+            :profile-image="profile_path || ''"
+            @get-actor-id="getActorDetails($event)"
+          />
+        </div>
+        <div
+          class="d-flex justify-content-end mt-2 seeMore"
+          @click="makeOptionalResponseAndRedirect('actors')"
+        >
+          see more...
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { mapActions, mapGetters, mapState } from "vuex";
+import SearchMovieCard from "@/components/Cards/SearchMovieCard";
+import SearchActorCard from "@/components/Cards/SearchActorCard";
+import SearchTvCard from "@/components/Cards/SearchTvCard";
+
+export default {
+  name: "SearchModal",
+  components: { SearchTvCard, SearchActorCard, SearchMovieCard },
+  methods: {
+    decreaseData(data) {
+      let arr = [];
```

Comment:
Make this method more declarative - return the modified results with `.filter` instead of modifying or copying array elements.